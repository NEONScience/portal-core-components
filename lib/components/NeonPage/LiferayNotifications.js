"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _styles = require("@material-ui/core/styles");
var _Snackbar = _interopRequireDefault(require("@material-ui/core/Snackbar"));
var _SnackbarContent = _interopRequireDefault(require("@material-ui/core/SnackbarContent"));
var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));
var _Close = _interopRequireDefault(require("@material-ui/icons/Close"));
var _Theme = _interopRequireWildcard(require("../Theme/Theme"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const useStyles = (0, _styles.makeStyles)(theme => ({
  notification: {
    color: theme.palette.text.primary,
    backgroundColor: _Theme.COLORS.GOLD[50],
    border: "1px solid ".concat(_Theme.COLORS.GOLD[300]),
    borderRadius: '4px',
    marginLeft: theme.spacing(3),
    [theme.breakpoints.up('sm')]: {
      maxWidth: '75%'
    },
    [theme.breakpoints.up('md')]: {
      maxWidth: '50%'
    },
    [theme.breakpoints.up('lg')]: {
      maxWidth: '33%'
    }
  }
}));
const LiferayNotifications = props => {
  const classes = useStyles(_Theme.default);
  const {
    notifications,
    onHideNotifications
  } = props;
  if (!notifications.length || notifications.every(n => n.dismissed)) {
    return null;
  }
  const renderNotificationContent = () => /*#__PURE__*/_react.default.createElement("div", {
    id: "neon-data-portal-notifications"
  }, notifications.map(notification => /*#__PURE__*/_react.default.createElement("div", {
    key: notification.id
    // eslint-disable-next-line react/no-danger
    ,
    dangerouslySetInnerHTML: {
      __html: notification.message
    }
  })));
  return /*#__PURE__*/_react.default.createElement(_Snackbar.default, {
    open: true,
    anchorOrigin: {
      vertical: 'bottom',
      horizontal: 'right'
    },
    ContentProps: {
      'aria-describedby': 'neon-data-portal-notifications'
    }
  }, /*#__PURE__*/_react.default.createElement(_SnackbarContent.default, {
    className: classes.notification,
    message: renderNotificationContent(),
    action: typeof onHideNotifications === 'function' ? /*#__PURE__*/_react.default.createElement(_IconButton.default, {
      color: "default",
      size: "small",
      key: "close",
      "aria-label": "close",
      onClick: onHideNotifications
    }, /*#__PURE__*/_react.default.createElement(_Close.default, null)) : null
  }));
};
LiferayNotifications.propTypes = {
  notifications: _propTypes.default.arrayOf(_propTypes.default.shape({
    id: _propTypes.default.string.isRequired,
    message: _propTypes.default.string.isRequired,
    dismissed: _propTypes.default.bool.isRequired
  })),
  onHideNotifications: _propTypes.default.func
};
LiferayNotifications.defaultProps = {
  notifications: [],
  onHideNotifications: null
};
var _default = exports.default = LiferayNotifications;