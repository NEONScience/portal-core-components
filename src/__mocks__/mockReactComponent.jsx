/* eslint react/prop-types: 0 */
/**
   Mock React Component

   Because jest.mock calls can't be dynamic / must hard-code an import path we still need individual
   mocks for every React component that requires mocking. This format defines a generic mock
   implementation for any react component that will appear in snapshots as a div with props mapped
   safely to attributes and any children intact.

   Usage:

   --

   import 'path/to/__mocks__/mockReactComponent';

   jest.mock('path/to/Component', () => mockReactComponent('path/to/Component'));
*/

import React from 'react';

export default function mockReactComponent(path) {
  return (props) => {
    const dataProps = {};
    Object.keys(props).forEach((prop) => {
      if (prop === 'children') { return; }
      const value = props[prop];
      let renderableValue = value;
      try {
        if (typeof value === 'object') { renderableValue = JSON.stringify(value); }
        if (typeof value === 'function') { renderableValue = value.toString(); }
      } catch (error) {
        renderableValue = `Non-renderable ${typeof value} (${error})`;
      }
      dataProps[`prop-${prop}`] = renderableValue;
    });
    const { children } = props;
    return children ? (
      <div ComponentPath={path} {...dataProps}>
        {children}
      </div>
    ) : (
      <div ComponentPath={path} {...dataProps} />
    );
  };
}
