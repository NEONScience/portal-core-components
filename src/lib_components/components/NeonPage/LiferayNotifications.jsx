import React from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from '@mui/styles';
import Snackbar from '@mui/material/Snackbar';
import SnackbarContent from '@mui/material/SnackbarContent';
import IconButton from '@mui/material/IconButton';

import CloseIcon from '@mui/icons-material/Close';

import Theme, { COLORS } from '../Theme/Theme';
import { resolveProps } from '../../util/defaultProps';

const useStyles = makeStyles((theme) => ({
  notification: {
    color: theme.palette.text.primary,
    backgroundColor: COLORS.GOLD[50],
    border: `1px solid ${COLORS.GOLD[300]}`,
    borderRadius: '4px',
    marginLeft: theme.spacing(3),
    [theme.breakpoints.up('sm')]: {
      maxWidth: '75%',
    },
    [theme.breakpoints.up('md')]: {
      maxWidth: '50%',
    },
    [theme.breakpoints.up('lg')]: {
      maxWidth: '33%',
    },
  },
}));

const defaultProps = {
  notifications: [],
  onHideNotifications: null,
};

const LiferayNotifications = (inProps) => {
  const props = resolveProps(defaultProps, inProps);
  const classes = useStyles(Theme);
  const { notifications, onHideNotifications } = props;

  if (!notifications.length || notifications.every((n) => n.dismissed)) { return null; }

  const renderNotificationContent = () => (
    <div id="neon-data-portal-notifications">
      {notifications.map((notification) => (
        <div
          key={notification.id}
          // eslint-disable-next-line react/no-danger
          dangerouslySetInnerHTML={{ __html: notification.message }}
        />
      ))}
    </div>
  );

  return (
    <Snackbar
      open
      anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
      ContentProps={{ 'aria-describedby': 'neon-data-portal-notifications' }}
    >
      <SnackbarContent
        className={classes.notification}
        message={renderNotificationContent()}
        action={(typeof onHideNotifications === 'function' ? (
          <IconButton
            color="default"
            size="small"
            key="close"
            aria-label="close"
            onClick={onHideNotifications}
          >
            <CloseIcon />
          </IconButton>
        ) : null)}
      />
    </Snackbar>
  );
};

LiferayNotifications.propTypes = {
  notifications: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      message: PropTypes.string.isRequired,
      dismissed: PropTypes.bool.isRequired,
    }),
  ),
  onHideNotifications: PropTypes.func,
};

export default LiferayNotifications;
