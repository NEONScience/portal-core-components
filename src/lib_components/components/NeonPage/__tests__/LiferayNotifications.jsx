import React from 'react';
import renderer from 'react-test-renderer';

import mockReactComponent from '../../../../__mocks__/mockReactComponent';

jest.mock('@mui/material/Snackbar', () => mockReactComponent('@mui/material/Snackbar'));

// eslint-disable-next-line import/first
import LiferayNotifications from '../LiferayNotifications';

describe('NeonPage - LiferayNotifications', () => {
  test('renders nothing if passed an empty array', () => {
    const tree = renderer.create(
      <LiferayNotifications />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders notifications with no actions if onHideNotifications not present', () => {
    const notifications = [
      { id: 'foo', message: 'bar', dismissed: false },
      { id: 'qux', message: '<b>HTML</b>', dismissed: true },
    ];
    const tree = renderer.create(
      <LiferayNotifications notifications={notifications} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders notifications with actions if onHideNotifications is present', () => {
    const notifications = [
      { id: 'foo', message: 'bar', dismissed: false },
      { id: 'qux', message: '<b>HTML</b>', dismissed: true },
    ];
    const tree = renderer.create(
      <LiferayNotifications notifications={notifications} onHideNotifications={() => {}} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
