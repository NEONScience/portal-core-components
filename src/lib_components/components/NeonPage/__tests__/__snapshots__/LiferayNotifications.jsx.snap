// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NeonPage - LiferayNotifications renders nothing if passed an empty array 1`] = `null`;

exports[`NeonPage - LiferayNotifications renders notifications with actions if onHideNotifications is present 1`] = `
<div
  ComponentPath="@material-ui/core/Snackbar"
  prop-ContentProps="{\\"aria-describedby\\":\\"neon-data-portal-notifications\\"}"
  prop-anchorOrigin="{\\"vertical\\":\\"bottom\\",\\"horizontal\\":\\"right\\"}"
  prop-open={true}
>
  <div
    className="MuiPaper-root MuiSnackbarContent-root makeStyles-notification-1 MuiPaper-elevation6"
    role="alert"
  >
    <div
      className="MuiSnackbarContent-message"
    >
      <div
        id="neon-data-portal-notifications"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "bar",
            }
          }
        />
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<b>HTML</b>",
            }
          }
        />
      </div>
    </div>
    <div
      className="MuiSnackbarContent-action"
    >
      <button
        aria-label="close"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`NeonPage - LiferayNotifications renders notifications with no actions if onHideNotifications not present 1`] = `
<div
  ComponentPath="@material-ui/core/Snackbar"
  prop-ContentProps="{\\"aria-describedby\\":\\"neon-data-portal-notifications\\"}"
  prop-anchorOrigin="{\\"vertical\\":\\"bottom\\",\\"horizontal\\":\\"right\\"}"
  prop-open={true}
>
  <div
    className="MuiPaper-root MuiSnackbarContent-root makeStyles-notification-1 MuiPaper-elevation6"
    role="alert"
  >
    <div
      className="MuiSnackbarContent-message"
    >
      <div
        id="neon-data-portal-notifications"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "bar",
            }
          }
        />
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<b>HTML</b>",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
